File: src/main/java/io/felipeandrade/gsw2/item/tool/GSWToolMaterial.java
package io.felipeandrade.gsw2.item.tool;

import net.minecraft.item.ItemConvertible;
import net.minecraft.item.ToolMaterial;
import net.minecraft.recipe.Ingredient;

public class GSWToolMaterial implements ToolMaterial {

    private final int miningLevel;
    private final int itemDurability;
    private final float miningSpeed;
    private final float attackDamage;
    private final int enchantability;
    private final Ingredient repairIngredient;

    GSWToolMaterial(int miningLevel, int itemDurability, float miningSpeed, float attackDamage,
                    int enchantability, Ingredient repairIngredient) {
        this.miningLevel = miningLevel;
        this.itemDurability = itemDurability;
        this.miningSpeed = miningSpeed;
        this.attackDamage = attackDamage;
        this.enchantability = enchantability;
        this.repairIngredient = repairIngredient;
    }

    public GSWToolMaterial(int miningLevel, int itemDurability, float miningSpeed, float attackDamage,
                           int enchantability, ItemConvertible... repairItems) {
        this(miningLevel, itemDurability, miningSpeed, attackDamage, enchantability,
                Ingredient.ofItems(repairItems));
    }

    public int getDurability() {
        return this.itemDurability;
    }

    public float getMiningSpeed() {
        return this.miningSpeed;
    }

    public float getAttackDamage() {
        return this.attackDamage;
    }

    public int getMiningLevel() {
        return this.miningLevel;
    }

    public int getEnchantability() {
        return this.enchantability;
    }

    public Ingredient getRepairIngredient() {
        return this.repairIngredient;
    }
}

File: gsw-mod/src/main/kotlin/io/felipeandrade/gsw/item/tool/GSWToolMaterial.kt
package io.felipeandrade.gsw.item.tool

import net.minecraft.item.ItemConvertible
import net.minecraft.item.ToolMaterial
import net.minecraft.recipe.Ingredient

class GSWToolMaterial internal constructor(
    private val miningLevel: Int,
    private val itemDurability: Int,
    private val miningSpeed: Float,
    private val attackDamage: Float,
    private val enchantability: Int,
    private val repairIngredient: Ingredient
) : ToolMaterial {
    constructor(
        miningLevel: Int, itemDurability: Int, miningSpeed: Float, attackDamage: Float,
        enchantability: Int, vararg repairItems: ItemConvertible
    ) : this(
        miningLevel, itemDurability, miningSpeed, attackDamage, enchantability,
        Ingredient.ofItems(*repairItems)
    )

    override fun getDurability(): Int = itemDurability
    override fun getMiningSpeedMultiplier(): Float = miningSpeed
    override fun getAttackDamage(): Float = attackDamage
    override fun getMiningLevel(): Int = miningLevel
    override fun getEnchantability(): Int = enchantability
    override fun getRepairIngredient(): Ingredient = repairIngredient
}